### ISSUE TITLE
gae.xml for compat images

### ISSUE THREAD

**Issue by lesv** [GoogleCloudPlatform/appengine-java-vm-runtime#198]
Should we document this generally?  @ludoch 

Where should we put it?  WEB-INF?  src/main/resources?

What is allowed in it?

**Comment by gregw** [comment]
@lesv 
Here is the current [gae.xml](https://github.com/GoogleCloudPlatform/appengine-java-vm-runtime/blob/async-support/jetty9-compat-base/src/main/jetty-base/etc/gae.xml].   This is not intended to be a user accessible file (although they can replace with an ADD in Dockerfile if they really want).

For most purposes it is hard coded, an in fact it currently has very few changeable parameters in it.  We could make more of the hard coded values there parameterised and put the default values into the [mod definition](https://github.com/GoogleCloudPlatform/appengine-java-vm-runtime/blob/async-support/jetty9-compat-base/src/main/jetty-base/modules/gae.mod), which end up in the ini file.  We would then have to document those parameters and how to change them.

So to summarise again - the jetty style of configuration (used by gae.xml) is that the XML is rarely edited or changed by users, but can be in extraordinary circumstances.   The majority of configuration is done by selecting which XMLs are used and by injecting parameters into them.

In compat, the XML is really just part of the module definition and we have [created modules](https://github.com/GoogleCloudPlatform/appengine-java-vm-runtime/tree/async-support/jetty9-compat-base/src/main/jetty-base/modules) for gae, gae-deploy and testMetaDataServer.  All of these are mostly transparent to the users an they are really just code in XML used to construct the compat server.  Users should rarely, if ever, need to change.

**Comment by lesv** [comment]
Thanks - just talked w/ @ludoch who said I shouldn't document this at that point.  I'm going to close it for now.
